beacon_node/lighthouse_network/src/service/utils.rs

```diff
@@ -37,32 +39,42 @@ pub struct Context<'a> {
 
 type BoxedTransport = Boxed<(PeerId, StreamMuxerBox)>;
 
-/// The implementation supports TCP/IP, WebSockets over TCP/IP, noise as the encryption layer, and
-/// mplex as the multiplexing layer.
+/// The implementation supports TCP/IP, QUIC (experimental) over UDP, noise as the encryption layer, and
+/// mplex/yamux as the multiplexing layer (when using TCP).
 pub fn build_transport(
     local_private_key: Keypair,
+    quic_support: bool,
 ) -> std::io::Result<(BoxedTransport, Arc<BandwidthSinks>)> {
+    // Creates the TCP transport layer
     let tcp = libp2p::tcp::tokio::Transport::new(libp2p::tcp::Config::default().nodelay(true));
+    // Enables DNS over the TCP transport.
     let transport = libp2p::dns::TokioDnsConfig::system(tcp)?;
-    #[cfg(feature = "libp2p-websocket")]
-    let transport = {
-        let trans_clone = transport.clone();
-        transport.or_transport(libp2p::websocket::WsConfig::new(trans_clone))
-    };
 
     // yamux config
     let mut yamux_config = yamux::Config::default();
     yamux_config.set_window_update_mode(yamux::WindowUpdateMode::on_read());
-    let (transport, bandwidth) = transport
+    let transport = transport
         .upgrade(core::upgrade::Version::V1)
         .authenticate(generate_noise_config(&local_private_key))
         .multiplex(yamux_config)
-        .timeout(Duration::from_secs(10))
-        .boxed()
-        .with_bandwidth_logging();
+        .timeout(Duration::from_secs(10));
+
+    let (transport, bandwidth) = if quic_support {
+        // Enables Quic
+        // The default quic configuration suits us for now.
+        let quic_config = libp2p_quic::Config::new(&local_private_key);
+        transport
+            .or_transport(libp2p_quic::tokio::Transport::new(quic_config))
+            .map(|either_output, _| match either_output {
+                Either::Left((peer_id, muxer)) => (peer_id, StreamMuxerBox::new(muxer)),
+                Either::Right((peer_id, muxer)) => (peer_id, StreamMuxerBox::new(muxer)),
+            })
+            .with_bandwidth_logging()
+    } else {
+        transport.with_bandwidth_logging()
+    };
 
-    // Authentication
-    Ok((transport, bandwidth))
+    Ok((transport.boxed(), bandwidth))
```

Comment:
`.with_bandwidth_logging` already calls `boxed`. Is the additional boxing needed here?

Reply:
nope, thanks Max addressed